### PR TITLE
Feature #66

### PR DESCRIPTION
**Pull request by Hammster**
# PULL REQUEST

Adding base functionality for #66 
## ADDITIONAL INFORMATION

Octicons are not properly displayed, the icon is aligned to the left, while the span and :before are centered

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ Settings | Type | Description | Default
 `gistId` | `String` | ID of the gist used as a backup storage. | `''`
 `setName` | `String` | Name of your working set, for example 'work' or 'home'. As each working set is backed up into a separate file in one gist, you can have multiple Client/Group/Project sets on different machines and have them all safely backed up on gist. | `default`
 `convertOldData` | `Boolean` | If you came from a version previous to <code>0.3.0</code>, you most probably have the old data in the atom folder. By default it will always check on startup for this data and if the new does not exist, it will convert to the new data schema. | `true`
+`iconListStyle` | `Boolean` | Displays the icon list without text, the text is still aviable on hover. | `false`
```
Comment (jccguimaraes) [inline README.md]: typo on `aviable`

File: src/form-view.js [hunk 2]
```diff
@@ -109,7 +109,12 @@ const viewMethods = {
     let iconsInput = document.createElement('atom-text-editor');
     iconsInput.setAttribute('mini', true);
     let iconsList = document.createElement('div');
-    iconsList.classList.add('block', 'icons-list');
+    let iconsListStyle = atom.config.get('project-viewer.iconListStyle');
```
Comment (jccguimaraes) [inline src/form-view.js]: can this be a `const` instead of a `let`?

File: src/config.js [hunk 3]
```diff
@@ -74,7 +74,13 @@ const config = {
         type: 'boolean',
         default: true,
         order: 10
-    }
+    },
+    'iconListStyle': {
+        description: 'Display only icons in update and create window (for more space).',
```
Comment (jccguimaraes) [inline src/config.js]: Does it make sense to be more like: `'Display as *only icons* in create/update view (for more space).'`?

File: src/form-view.js [hunk 4]
```diff
@@ -141,7 +146,14 @@ const viewMethods = {
         icon.pack,
         `${icon.prefix}${icon.name}${icon.suffix}`
       );
-      iconView.textContent = icon.description;
+
+      if (iconsListStyle) {
+        iconView.textContent = "";
+        this.disposables.add(atom.tooltips.add(iconView, {title: icon.description}));
```
Comment (jccguimaraes) [inline src/form-view.js]: there is a property in the object options that you can change the timeouts on show and hide, `{title: icon.description, delay: {show: 100, hide: 100}}`

File: src/form-view.js (hunk 2 above)
Comment (jccguimaraes) [inline src/form-view.js]: Also try using the `atom.confid.onDidChange('project-viewer.iconListStyle', (newValue) => {})` to change opened panels from this style

Comment (Hammster) [inline src/form-view.js]: All rewrite this part then, even though after i add the `onDidChange` it is not constant and therefore a `let` ;)

File: src/form-view.js (hunk 4 above)
Comment (Hammster) [inline src/form-view.js]: yeah maybe 1/2th or 1/3th of the time would be good, i'll change that.

File: src/form-view.js [hunk 5]
```diff
@@ -109,7 +109,12 @@ const viewMethods = {
     let iconsInput = document.createElement('atom-text-editor');
     iconsInput.setAttribute('mini', true);
     let iconsList = document.createElement('div');
-    iconsList.classList.add('block', 'icons-list');
+    let iconsListStyle = atom.config.get('project-viewer.iconListStyle');
+    if (iconsListStyle) {
+      iconsList.classList.add('block', 'icons-list', 'only-icons');
+    } else {
+      iconsList.classList.add('block', 'icons-list');
+    }
```
Comment (jccguimaraes) [inline src/form-view.js]: better yet, do this instead (in my opinion):

```
iconsList.classList.add('block', 'icons-list');
this.disposables.add(
    atom.config.observe('project-viewer.iconListStyle', (newValue) => {
        iconsList.classList.toggle('only-icons', newValue);
    })
);
...
```

File: src/form-view.js (hunk 2 above)
Comment (jccguimaraes) [inline src/form-view.js]: instead of `atom.confid.onDidChange` check next comment

**Comment by jccguimaraes** [comment]
hi @Hammster,
Sorry for the mess, but this needs another review... a lot has changed and will still change

**Pull request closed without merging**